wordpress-mobile/WordPress-iOS
# Shared Login: Replace old keychain calls, copy keychain, and copy database to shared directory

## Description

- Replaces existing calls to `SFHFKeychainUtils` with `KeychainUtils`
- Copies the local keychain to the shared keychain on app launch
- Copies the local database to the shared directory on `init` for `ContextManager`

## Testing

The keychain sharing doesn't seem to work quite right on the simulator so I was using a real device to test. On the simulator, the access group seems to be ignored. Everything _should_ still work on a simulator though.

To test:
- Fresh install WordPress with the feature flag set to `false`
- Login
- Tap on your avatar > App Settings > Debug > Enable `Shared Login`
- Close and relaunch WordPress
- Verify you are still logged in
- Close WordPress
- Set the feature flag `sharedLogin` to `true` in `FeatureFlag.swift`
- Fresh install Jetpack
- Verify you are logged in without needing to login
- Log out from Jetpack
- Close Jetpack
- Open WordPress
- Verify you are logged out

**Note:** There are some strange behaviors if you keep both apps open in memory and logout from one or the other. We will probably have to address what to do in this scenario in a future PR.

## Regression Notes
1. Potential unintended areas of impact
Anywhere that uses the keychain or core data

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually testing, running unit tests, it'll require further extensive testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
